airbyte-webapp/src/views/Connector/ServiceForm/useAuthentication.tsx

```diff
@@ -0,0 +1,210 @@
+ * parameter values, and joins them to string paths.
+ */
+const convertAndPrefixPaths = (paths?: Path[], rootPath: Path = []): string[] => {
+  return (
+    paths?.map((pathParts) => {
+      return makeConnectionConfigurationPath([...stripNumericalEntries(rootPath), ...stripNumericalEntries(pathParts)]);
+    }) ?? []
+  );
+};
+
+/**
+ * Returns true if the auth button should be shown for a advancedAuth specification.
+ * This will check if the connector has a predicateKey, and if so, check if the current form value
+ * of the corresponding field matches the predicateValue from the specification.
+ */
+const shouldShowButtonForAdvancedAuth = (
+  predicateKey: string[] | undefined,
+  predicateValue: string | undefined,
+  values: ServiceFormValues<unknown>
+): boolean => {
+  return (
+    !predicateKey ||
+    predicateKey.length === 0 ||
+    predicateValue === getIn(values, makeConnectionConfigurationPath(predicateKey))
+  );
+};
+
+/**
+ * Returns true if the auth button should be shown for a authSpecification connector.
+ */
+const shouldShowButtonForLegacyAuth = (
+  spec: JSONSchema7,
+  rootPath: Path,
+  values: ServiceFormValues<unknown>
+): boolean => {
+  if (!rootPath.some((p) => isNumerical(p))) {
+    // If the root path of the auth parameters (which is also the place the button will be rendered)
+    // is not inside a conditional, i.e. none of the root path is a numerical value, we will always
+    // show the button.
+    return true;
+  }
+
+  // If the spec had a root path inside a conditional, e.g. `credentials.0`, we need to figure
+  // out if that conditional is currently on the correct selected option. Since the connector spec
+  // in contrast to `advancedAuth`, which has `predicateValue` doesn't have the value for the conditional
+  // we need to find that ourselves first.
+
+  // To find the path inside the connector spec that matches the `rootPath` we'll need to insert `properties`
+  // and `oneOf`, since they'll appear in the JSONSchema, e.g. this turns `credentials.0` to `properties.credentials.oneOf.0`
+  const specPath = rootPath.flatMap((path) =>
+    isNumerical(path) ? ["oneOf", String(path)] : ["properties", String(path)]
+  );
+  // Get the part of the spec that `rootPath` point to
+  const credentialsSpecRoot = getIn(spec, specPath) as JSONSchema7 | undefined;
+
+  if (!credentialsSpecRoot?.properties) {
+    // if the part doesn't exist in the spec (which should not happen) we just show the auth button always.
+    return true;
+  }
+
+  // To find the value we're expecting, we run through all properties inside that matching spec inside the conditional
+  // to find the one that has a `const` value in it, since this is the actual value that will be written into the conditional
+  // field itself once it's selected.
+  const constProperty = Object.entries(credentialsSpecRoot.properties)
+    .map(([key, prop]) => [key, typeof prop !== "boolean" ? prop.const : undefined] as const)
```

Comment:
I'm forgetting exactly why you added the whole `typeof prop !== "boolean"` piece, and the `as const` piece here. Since this code should be going away soon anyway its probably not super important to have a detailed comment explaining this, but I am curious for my own learning

Reply:
Huh, weird. The `| boolean` on that type doesn't make much sense to me either 🤷 but at least your code makes sense to me then